Notifications - needs enable flag possibility

Limiria opened the pull request:
On the notifications branch is now a version where all players on the players infopage will get a notification if someone presses the report button. It's to mention that I only have tested this feature on one device with 2 different tabs in Firefox and one Opera tab. I would test it again on mobile devices, because I don't know exactly how they would behave in this case.

Limira00 committed:
Added the request to allow notifications on index.html + added new flask

Limira00 committed:
Now a report Notification will send out a Notification on report to all
players

Nimuedaertya commented on src/endpoints.py:
```diff
@@ -26,6 +27,28 @@
 assets = Environment(app)
 css = Bundle(const.PATH_CSS_INPUT, output=const.PATH_CSS_OUTPUT)
+socketio = SocketIO(app)
+app.config['SECRET_KEY'] = 'secret!'
```
constants should be moved to the const file. Secrets however should not be pushed

Nimuedaertya reviewed (changes requested):
Looks good, tested it locally so far.

Limira00 committed:
Made flask on hosting hopefully a "secure context"

- Remove static cert, add to init.sh

Commit:
cleanups

Motivation
  - certs should not lie in root directory;
    moved to config file directory
  - move 'secret' to static variables

Commit:
Rebase onto main, adding mising dependency

Commit:
linting fix

Nimuedaertya commented:
fixes #15

Commit:
[ FEAT ] Dualstack http and https possible

Motivation
  - flag to toggle between http and https (-s / --secure)
  - changed behavior for python based on flag
  - MISSING: Templating is acting not differently based on that
